### ISSUE TITLE
flutter devices doesn't detect Xcode 8 simulators when Xcode 9 command-line tools are selected

### ISSUE THREAD

**Issue by cbracken** [flutter/flutter#12345]
## Steps to Reproduce

1. Open an Xcode 8 iPhone simulator.
1. Select the Xcode 9 command-line tools: `sudo xcode-select -s /Applications/Xcode9.0.app/Contents/Developer`
1. Run `flutter devices`.

No simulators are listed and no error message is emitted. If Xcode's command-line tools don't recognise simulators from other Xcode versions, I wonder if it would be a good idea for us to:
1. Detect multiple Xcode installations and emit a log line with the current Xcode version.
2. Emit the fact that multiple Xcode installations exist in `flutter doctor` to help us debug cases like this.

## Logs

```
[  +17 ms] [/Users/cbracken/src/flutter/flutter/] git rev-parse --abbrev-ref --symbolic @{u}
[  +21 ms] Exit code 0 from: git rev-parse --abbrev-ref --symbolic @{u}
[        ] origin/master
[        ] [/Users/cbracken/src/flutter/flutter/] git ls-remote --get-url origin
[   +4 ms] Exit code 0 from: git ls-remote --get-url origin
[        ] git@github.com:cbracken/flutter.git
[        ] [/Users/cbracken/src/flutter/flutter/] git log -n 1 --pretty=format:%H
[   +5 ms] Exit code 0 from: git log -n 1 --pretty=format:%H
[        ] 89566feeed2293f709cab723a970af617adf042f
[        ] [/Users/cbracken/src/flutter/flutter/] git log -n 1 --pretty=format:%ar
[   +5 ms] Exit code 0 from: git log -n 1 --pretty=format:%ar
[        ] 2 days ago
[ +150 ms] /usr/bin/defaults read /Applications/Android Studio.app/Contents/Info CFBundleShortVersionString
[  +43 ms] Exit code 0 from: /usr/bin/defaults read /Applications/Android Studio.app/Contents/Info CFBundleShortVersionString
[        ] 2.3
[ +104 ms] Listing devices using /Users/cbracken/Library/Android/sdk/platform-tools/adb
[        ] /Users/cbracken/Library/Android/sdk/platform-tools/adb devices -l
[  +14 ms] Exit code 0 from: /Users/cbracken/Library/Android/sdk/platform-tools/adb devices -l
[        ] List of devices attached
[   +3 ms] idevice_id -h
[ +178 ms] /usr/bin/xcrun simctl list --json devices
[ +100 ms] No devices detected.

                   If you expected your device to be detected, please run "flutter doctor" to diagnose
                   potential issues, or visit https://flutter.io/setup/ for troubleshooting tips.
[  +12 ms] "flutter devices" took 554ms.
[ +148 ms] ensureAnalyticsSent: 145ms
[        ] exiting with code 0
```
## Flutter Doctor

```
[✓] Flutter (on Mac OS X 10.13 17A365, locale en-AU, channel master)
    • Flutter at /Users/cbracken/src/flutter/flutter
    • Framework revision 89566feeed (2 days ago), 2017-09-29 17:56:25 -0700
    • Engine revision bdfedcc26f
    • Tools Dart version 1.25.0-dev.11.0

[✓] Android toolchain - develop for Android devices (Android SDK 25.0.3)
    • Android SDK at /Users/cbracken/Library/Android/sdk
    • Platform android-25, build-tools 25.0.3
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_112-release-b06)

[✓] iOS toolchain - develop for iOS devices (Xcode 9.0)
    • Xcode at /Applications/Xcode9.0.app/Contents/Developer
    • Xcode 9.0, Build version 9A235
    • ios-deploy 1.9.2
    • CocoaPods version 1.3.1

[✓] Android Studio (version 2.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Java version OpenJDK Runtime Environment (build 1.8.0_112-release-b06)

[✓] IntelliJ IDEA Community Edition (version 2017.2.5)
    • Flutter plugin version 17.0
    • Dart plugin version 172.4155.35
```

**Comment by giraldeau** [comment]
I have a similar issue after upgrading to Xcode 9.0.1. The devices are not listed anymore, even though `open -a Simulator.app` actually starts the iOS device. I ran the command `xcode-select` to make sure that the current Xcode is the default. All devices returned by `/usr/bin/xcrun simctl list --json devices | grep state` are in the Shutdown state, so that looks like an issue with Xcode rather than flutter itself.

**Comment by cbracken** [comment]
@giraldeau thanks for the doctor output. 

Given that Xcode itself doesn't support this, the fix here is for us to emit an actionable `flutter doctor` message to steer the user in the right direction.

**Comment by giraldeau** [comment]
@cbracken thanks for the feedback. I understand, the fault is not on flutter, but do we have a workaround for this problem, or is it something to investigate? Thanks!

**Comment by cbracken** [comment]
There are a couple things we can do here I think:
1. If multiple Xcode installations are present, emit that in `flutter doctor`. A reasonable assumption is to check the `/Applications` and `$HOME/Applications` directories for directories named `Xcode*.app` that contain `Contents/Developer/usr/bin/xcodebuild`.
2. As an improvement on (1) get the version info from each of those and if the currently `xcode-select`ed version is not the latest, warn.

For (2), ideally we'd probably filter betas, though I'm not sure. I ran into an issue today where after testing with Xcode 9.1beta2, `xcrun simctl list --json` shows that my existing iOS 11.0 simulators were marked `"availability" : " (unavailable, runtime profile not found)",`. 

This is the first time I've run into an issue where I've been unable to downgrade Xcode versions smoothly, so could be an issue with the beta, but something to think about when we address this.

**Comment by giraldeau** [comment]
The problem is solved with Xcode 9.2, Build version 9C40b.

**Comment by cbracken** [comment]
@giraldeau thanks for updating! 

I've also landed #13622 to reduce the chance of this in the future. Since August 2017, Flutter has required Xcode 9 for engine development and builds, and since a week ago we've required Xcode 9 for app builds. That change does not affect the minimum supported iOS deployment target, which remains iOS 8. I think with the `flutter doctor` change in place, we're probably safe to close this for now.
